File: 0424 longest-repeating-character-replacement/solution.java
class Solution {
    public int characterReplacement(String s, int k) {
        int l=0,maxf=0,maxl=0;
        int[] map = new int[126];
        for(int r=0;r<s.length();r++) {
            char c = s.charAt(r);
            map[c]++;
            maxf=Math.max(maxf,map[c]);
            while((r-l+1) - maxf > k ) {
                map[s.charAt(l)]--;
                l++;
            }
            maxl=Math.max(r-l+1,maxl);
        }
        return maxl;
    }
}

File: 0424 longest-repeating-character-replacement/solution.go
func characterReplacement(s string, k int) int {
    l,mxl,mxf:=0,0,0
    var arr [126]int

    for r,v := range s {
        arr[v]++
        mxf=max(mxf,arr[v])
        for (r-l+1)-mxf > k {
            arr[s[l]]--
            l++
        }
        mxl=max(mxl,r-l+1)
    }
    return mxl
}
